File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dic = {}
    for i in clothes:
        if i[1] in dic:
            dic[i[1]].append(i[0])
        else:
            dic[i[1]] = [i[0]]

    answer = 1
    for _, value in dic.items():
        answer *= (len(value) + 1)

    return answer-1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> map = new HashMap<>();

        for (String[] clothe : clothes) {
            map.put(clothe[1], map.getOrDefault(clothe[1], 0) + 1);
        }

        for (Map.Entry<String, Integer> entry : map.entrySet()) {
            Integer value = entry.getValue();
            answer *= (value + 1);


        }
        return answer - 1;
    }
}